statsmodels/discrete/tests/test_discrete.py

```diff
@@ -2377,3 +2377,31 @@ def test_unchanging_degrees_of_freedom():
     # Test that the call to `fit_regularized` didn't modify model.df_model inplace.
     assert_equal(res3.df_model, res1.df_model)
     assert_equal(res3.df_resid, res1.df_resid)
+
+
+@pytest.mark.parametrize('method', ['newton', 'bfgs'])
+@pytest.mark.parametrize('cov_type', ['nonrobust', 'HC0', 'HC1', 'HC2', 'HC3'])
+@pytest.mark.parametrize('use_t', [True, False])
+@pytest.mark.parametrize('use_transparams', [True, False])
+def test_negbinp_cov_types(use_transparams, use_t, cov_type, method):
+    # GH#5234 Test that cov_type and use_t are correctly attached to the
+    # results of NegativeBinomialP.fit
+
+    # avoid runtime warning in cases where use_transparams is ignored
+    use_transparams = use_transparams and method not in ['newton', 'ncg']
+
+    # Data construction based
+    # on TestNegativeBinomialPL1Compatability.setup_class
+    rand_data = sm.datasets.randhie.load(as_pandas=False)
+
+    rand_exog = rand_data.exog.view(float).reshape(len(rand_data.exog), -1)
+    rand_exog_st = (rand_exog - rand_exog.mean(0)) / rand_exog.std(0)
+    rand_exog = sm.add_constant(rand_exog_st, prepend=True)
+
+    model = sm.NegativeBinomialP(rand_data.endog, rand_exog)
+    result = model.fit(method=method, cov_type=cov_type, use_t=use_t,
+                       use_transparams=use_transparams,
+                       disp=False)
+
+    assert result.use_t is use_t
```

Comment:
All four of the `use_t = True`, `cov_type='nonrobust'` cases fail under master.  @josef-pkt is that intentional?

Reply:
@josef-pkt insights here?

Reply:
@josef-pkt when you get a chance this is a specifically-your-attention-needing thing.  Is the existing behavior intentional?

Reply:
This might be a bug as in #5331

Reply:
> This might be a bug as in #5331

Neat.  Before I dig into this, is the simplification+unification in this PR something you're open to?